### PR TITLE
Adds option to determine Split Vectors output format [needs-docs]

### PR DESCRIPTION
**Pull request by SrNetoChan**
## Description

It adds a new string parameter to the Split Vector Layer algorithm
This allows the user to select the format of the output files.
The default is GeoPackage. But other formats are available.

## Checklist

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [ ] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### CODE REVIEW

**Comment by SrNetoChan** [comment]
@alexbruy can you please review? thanks!

**Comment by DelazJ** [comment]
@SrNetoChan it Should be [needs-docs] to generate a report

**Review by alexbruy — COMMENTED** [review]
In general I will be in favor of more generic solution, allowing to change output format also for other similar tools. But as temporary solution it looks good enough.

File: python/plugins/processing/algs/qgis/VectorSplit.py [hunk 1]
```diff
@@ -60,10 +63,16 @@ def __init__(self):
     def initAlgorithm(self, config=None):
         self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
                                                               self.tr('Input layer')))
-
         self.addParameter(QgsProcessingParameterField(self.FIELD,
                                                       self.tr('Unique ID field'), None, self.INPUT))
-
+        self.FORMATS = OrderedDict([('Geopackage', 'gpkg'),
```
[the remark this answers is not in the document]
Comment (SrNetoChan) [reply python/plugins/processing/algs/qgis/VectorSplit.py]: Not sure if all vector formats would be supported as output format just by the extension. Wouldn't more settings need to be added for specific formats?

**Comment by SrNetoChan** [comment]
> @SrNetoChan it Should be [needs-docs] to generate a report

Shame on me...
I changed de description. Since I will add more commits and squash in the end, that should provide a good commit message in the end. Right?

File: python/plugins/processing/algs/qgis/VectorSplit.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SrNetoChan) [reply python/plugins/processing/algs/qgis/VectorSplit.py]: > In general I will be in favor of more generic solution, allowing to change output format also for other similar tools. But as temporary solution it looks good enough.

Do you mean a new parameter type for the processing framework?

**Comment by nyalldawson** [comment]
> But as temporary solution it looks good enough.

My only reservation is that this isn't a temporary solution -- it becomes part of our stable API which we'll have to carry around forever, even if a proper approach to handling this is later added.

**Comment by SrNetoChan** [comment]
What would be a proper way of handlinhg this? Would it need to be implemented in C++ or python? With some pointers, I could try to work on a proper approach, whatever we decide that would be. 

The reason why I proposed this extra option was because I think it's extremely valuable for users.

**Comment by nyalldawson** [comment]
I think I needs to be added to QgsProcessingContext, as a "preferredVectorFormat" option. 

That bits easy. The tricky bit is that we don't yet have any way of exposing this choice as a per algorithm execution. We need to extend the processing dialog, adding options for setting the context (eg this preferred setting, an "extent filter", transformation settings, an option to override the invalid feature handling, etc).

I guess for now we could add it to the context, and then hook in the existing global processing setting to the context?

**Comment by SrNetoChan** [comment]
> I think I needs to be added to QgsProcessingContext, as a "preferredVectorFormat" option.
> 
> That bits easy. The tricky bit is that we don't yet have any way of exposing this choice as a per algorithm execution. We need to extend the processing dialog, adding options for setting the context (eg this preferred setting, an "extent filter", transformation settings, an option to override the invalid feature handling, etc).
> 
> I guess for now we could add it to the context, and then hook in the existing global processing setting to the context?

Uhmm. This looks to be too much for my "skills".

**Comment by gioman** [comment]
> Uhmm. This looks to be too much for my "skills".

this patch solves a very real/practical issue, despite being not "sub optimal", can't be merged as it is?

**Comment by nyalldawson** [comment]
@gioman 
I'm on it, don't merge.

**Pull request closed without merging**